app/controllers/katello/api/v2/systems_bulk_actions_controller.rb

```diff
@@ -154,7 +155,9 @@ def environment_content_view
     param :errata_ids, Array, :desc => N_("List of Errata ids")
     def available_incremental_updates
       version_environments = {}
-      systems = System.with_non_installable_errata(@errata).where("#{System.table_name}.id" => @systems)
+      host_ids = @systems.pluck(:host_id)
+      systems = Katello::Host::ContentFacet.with_non_installable_errata(@errata).
+          where("#{Katello::Host::ContentFacet.table_name}.host_id" => host_ids)
 
       ContentViewEnvironment.for_systems(systems).each do |cve|
```

Comment:
This may work, but you probably wanna write a new method for_content_facet.  I've already written here for my PR:
```
    def self.for_content_facet(content_facet)
      joins("INNER JOIN #{Host::ContentFacet.table_name} on #{Host::ContentFacet.table_name}.lifecycle_environment_id = #{ContentViewEnvironment.table_name}.environment_id").
           where("#{Host::ContentFacet.table_name}.content_view_id = #{ContentViewEnvironment.table_name}.content_view_id").where("#{Host::ContentFacet.table_name}.id" => content_facet).uniq
    end
```